IndexedDB/idbtransaction_objectStoreNames.html

```diff
@@ -0,0 +1,188 @@
+            'transaction objectStoreNames should have new store');
+
+        db.createObjectStore('s2');
+        assert_array_equals(db.objectStoreNames, ['s1', 's2', 's3'],
+            'database should have three stores');
+        assert_array_equals(tx.objectStoreNames, ['s1', 's2', 's3'],
+            'transaction objectStoreNames should be sorted');
+
+        db.deleteObjectStore('s1');
+        assert_array_equals(db.objectStoreNames, ['s2', 's3'],
+            'database should have two stores');
+        assert_array_equals(tx.objectStoreNames, ['s2', 's3'],
+            'transaction objectStoreNames should be updated after delete');
+    });
+    open.onsuccess = t.step_func(function() {
+        var db = open.result;
+        db.close();
+
+        assert_array_equals(db.objectStoreNames, ['s2', 's3'],
+            'connection should have snapshot of store names after close');
+        assert_array_equals(tx.objectStoreNames, ['s2', 's3'],
+            'transaction should have snapshot of store names after close');
+
+        var open2 = indexedDB.open(dbname, 2);
+        open2.onerror = t.unreached_func('open should succeed');
+        open2.onupgradeneeded = t.step_func(function() {
+            var db2 = open2.result;
+            var tx2 = open2.transaction;
+            assert_array_equals(db2.objectStoreNames, ['s2', 's3'],
+                'database should have two stores');
+            assert_array_equals(tx2.objectStoreNames, ['s2', 's3'],
+                'transaction should have two stores in scope');
+
+            db2.createObjectStore('s4');
+            assert_array_equals(db2.objectStoreNames, ['s2', 's3', 's4'],
+                'database should have three stores');
+            assert_array_equals(tx2.objectStoreNames, ['s2', 's3', 's4'],
+                'transaction should have new store in scope');
+
+            assert_array_equals(db.objectStoreNames, ['s2', 's3'],
+                'previous connection objectStoreNames should be unchanged');
+            assert_array_equals(tx.objectStoreNames, ['s2', 's3'],
+                'previous transaction objectStoreNames should be unchanged');
+
+            t.done();
+        });
+    });
+
+}, 'IDBTransaction.objectStoreNames in upgrade transactions');
+
+async_test(function(t) {
+    var dbname = document.location + '-' + t.name;
+    var del = indexedDB.deleteDatabase(dbname);
+    del.onerror = t.unreached_func('deleteDatabase should succeed');
+    var open = indexedDB.open(dbname, 1);
+    open.onerror = t.unreached_func('open should succeed');
+
+    open.onupgradeneeded = t.step_func(function() {
+        var db = open.result;
+        assert_array_equals(db.objectStoreNames, [],
+            'database should have no stores');
+        db.createObjectStore('s1');
+        db.createObjectStore('s2');
+        db.createObjectStore('s3');
+        assert_array_equals(db.objectStoreNames, ['s1', 's2', 's3'],
```

Comment:
Can you clarify what you mean here?

(These assertions about db.objectStoreNames really shouldn't be part of the test anyway, since this test not about that attribute. But I don't think that's what you mean.)